### PR TITLE
don't auto-enable use of `-DSCOTCH_PTHREAD` when using MPI library other than Intel MPI in SCOTCH easyblock

### LINKED ISSUE #2972: Threaded mpi should be disabled by default in SCOTCH for all toolchains
https://github.com/easybuilders/easybuild-easyblocks/blob/2ecd3c97fa1563c7634adf4467f5081fd2debca1/easybuild/easyblocks/s/scotch.py#L112-L115

Comment indicates that it should have been dropped a long time ago. Lets do it for EB 5.0

### PR DESCRIPTION
**Pull request by Micket**
Should have been done for EB 3.0 according to the comment in the script. No need to convert any old easyconfigs back into the old behavior to try and preserve anything, because old behavior is bad; i'm pretty sure we want this. 

cf. in https://github.com/easybuilders/easybuild-easyconfigs/pull/17459#discussion_r1268199859

Fixes #2972

### CODE REVIEW

**Comment by boegel** [comment]
see also discussion in #914

**Comment by Micket** [comment]
Test report by @Micket
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _SCOTCH-7.0.3-gompi-2023a.eb_ 

Build succeeded for 1 out of 1 (1 easyconfigs in total)
alvis-c1 - Linux Rocky Linux 8.8, x86_64, Intel Xeon Processor (Skylake), Python 3.6.8
See https://gist.github.com/Micket/62bc193be0f01073c29e9a32aafdbf75 for a full test report.

**Comment by Micket** [comment]
Test report by @Micket
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _SCOTCH-7.0.3-gompi-2022b.eb_ 

Build succeeded for 1 out of 1 (1 easyconfigs in total)
alvis-c1 - Linux Rocky Linux 8.8, x86_64, Intel Xeon Processor (Skylake), Python 3.6.8
See https://gist.github.com/Micket/5572fe430a4c18bcbb843d72bec2f962 for a full test report.

**Review by casparvl — APPROVED** [review]
Lgtm! I'm just checking with other maintainers how the PRs that target 5.0 are handled, before I press the merge button. Want to make extra sure I don't create an unwanted situation where 5.0 stuff ends up in a 4.X release ;-)

**Comment by casparvl** [comment]
Ok, all good, I see this targets the 5.0.X branch. Good to go!

**Pull request merged**
